cpp/src/arrow/util/bit-util.h

```diff
@@ -68,9 +68,13 @@ static inline void SetBit(uint8_t* bits, int64_t i) {
 }
 
 static inline void SetBitTo(uint8_t* bits, int64_t i, bool bit_is_set) {
-  // See https://graphics.stanford.edu/~seander/bithacks.html
+  // TODO: speed up. See https://graphics.stanford.edu/~seander/bithacks.html
   // "Conditionally set or clear bits without branching"
-  bits[i / 8] ^= static_cast<uint8_t>(-bit_is_set ^ bits[i / 8]) & kBitmask[i % 8];
+  if (bit_is_set) {
```

Comment:
I couldn't figure out how to port the bit twiddling code in a way that MSVC was happy about. Since we're only using this function in `CopyBitmap` right now, I left a TODO and figured we can return to it later